streams/writable-streams/properties.js

```diff
@@ -194,28 +194,32 @@ for (const method in sinkMethods) {
     });
   }, `sink method ${method} should be called even when it's located on the prototype chain`);
 
-  if (method !== 'start') {
-    promise_test(t => {
-      const unreachedTraps = ['getPrototypeOf', 'setPrototypeOf', 'isExtensible', 'preventExtensions',
-                              'getOwnPropertyDescriptor', 'defineProperty', 'has', 'set', 'deleteProperty', 'ownKeys',
-                              'apply', 'construct'];
-      const handler = {
-        get: t.step_func((target, property) => {
-          if (property === 'type') {
-            return undefined;
-          }
-          assert_in_array(property, ['start', method], `only start() and ${method}() should be called`);
-          return () => Promise.resolve();
-        })
-      };
-      for (const trap of unreachedTraps) {
-        handler[trap] = t.unreached_func(`${trap} should not be trapped`);
-      }
-      const sink = new Proxy({}, handler);
-      const ws = new WritableStream(sink);
-      return trigger(ws.getWriter());
-    }, `unexpected properties should not be accessed when calling sink method ${method}`);
-  }
+  promise_test(t => {
+    const unreachedTraps = ['getPrototypeOf', 'setPrototypeOf', 'isExtensible', 'preventExtensions',
+                            'getOwnPropertyDescriptor', 'defineProperty', 'has', 'set', 'deleteProperty', 'ownKeys',
+                            'apply', 'construct'];
+    const touchedProperties = [];
+    const handler = {
+      get: t.step_func((target, property) => {
+        touchedProperties.push(property);
+        if (property === 'type') {
+          return undefined;
+        }
+        return () => Promise.resolve();
+      })
+    };
+    for (const trap of unreachedTraps) {
+      handler[trap] = t.unreached_func(`${trap} should not be trapped`);
+    }
+    const sink = new Proxy({}, handler);
+    const ws = new WritableStream(sink);
+    assert_array_equals(touchedProperties, ['type', 'write', 'close', 'abort', 'start'],
+                        'expected properties should be got');
+    return trigger(ws.getWriter()).then(() => {
+      assert_array_equals(touchedProperties, ['type', 'write', 'close', 'abort', 'start'],
+                          'no properties should be accessed on method call');
+    });
+  }, `unexpected properties should not be accessed when calling sink method ${method}`);
```

Comment:
Should we clone this test for readable too, while we're here?

(What a great test, BTW...)

Reply:
I've been thinking about throwing away readable-stream's properties.js and use one based on this instead. But I'd have to check carefully that I wasn't losing any coverage, which is why I haven't done it yet.